dashboard/src/components/injector/che-injector.service.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2015-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+
+/**
+ * This class is handling the service for the additional element injection.
+ * @author Oleksii Orel
+ */
+export class CheInjectorService {
+
+  /**
+   * Default constructor that is using resource injection
+   * @ngInject for Dependency injection
+   */
+  constructor($timeout, $document, $compile) {
+    this.$timeout = $timeout;
+    this.$document = $document;
+    this.$compile = $compile;
+
+    //object of destination elements with id as keys and innerHTML of additional widget as value
+    //{ parentElementId: { additionalElementId: additionalElementInnerHTML } }
+    this.additionalElements = {};
```

Comment:
elementsToInject  
To me, it is understandable.
Done

In this case I prefer work with the object